tailnet/controllers.go

```diff
@@ -922,14 +960,13 @@ type tunnelUpdater struct {
 	client         WorkspaceUpdatesClient
 	coordCtrl      *TunnelSrcCoordController
 	dnsHostsSetter DNSHostsSetter
+	updateCallback func(*proto.WorkspaceUpdate)
```

Comment:
I strongly prefer an interface for this sort of thing.

Firstly, I have a suspicion that the interactions with the `vpn.Tunnel` will grow with time, leading to more methods.

Secondly, even with a single function, an interface makes it much easier to figure out what's on the other side because you can "Show implementations" in your IDE. One or more will be "fake" or "mock" and so it's pretty clear what is on the other side in production. With anonymous function callbacks, you have to manually chase the call chain.